### PR TITLE
Use fallback approach for nested matmul

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85311

### CODE REVIEW

**Review by cpuhrsch — APPROVED** [review]

File: aten/src/ATen/native/nested/NestedTensorMath.cpp [hunk 1]
```diff
@@ -909,8 +909,8 @@ Tensor matmul_nested(const Tensor& self, const Tensor& mat2) {
     if (batch_size == 1) {
       at::mm_out(
           output_unbind[i],
-          self_buffer.as_strided(self_size, self_strides[i], self_offsets[i]),
-          mat2_buffer.as_strided(mat2_size, mat2_strides[i], mat2_offsets[i])
+          self_buffer.as_strided(self_size, self_strides[i], self_offsets[i]).squeeze(0),
```
Comment (cpuhrsch) [inline aten/src/ATen/native/nested/NestedTensorMath.cpp:912]: Wait, doesn't this *require* the input to be 4-dim? What happens if you pass a 5-dim NT?

In general, let's replace these for loops by a call to `to_padded`, `matmul`, `from_padded` until we have a more performant kernel. All these kernel launches are very inefficient.

**Comment by drisspg** [comment]
fixes: #85064

File: aten/src/ATen/native/nested/NestedTensorMath.cpp (hunk 1 above)
Comment (mikaylagawarecki) [reply aten/src/ATen/native/nested/NestedTensorMath.cpp:912]: Oh you are right.

File: aten/src/ATen/native/nested/NestedTensorMath.cpp [hunk 2]
```diff
@@ -864,9 +864,12 @@ Tensor matmul_nested(const Tensor& self, const Tensor& mat2) {
   else if (!self.is_nested() && mat2.is_nested()) {
     AT_ERROR("Expected both to be nested, but got a non-nested self and nested other");
   }
+  // to_padded_tensor only supports contiguous inputs
```
Comment (mikaylagawarecki) [inline aten/src/ATen/native/nested/NestedTensorMath.cpp:867]: Just calling out that these additional contig calls are necessary, we do use matmul with noncontig inputs in the at::sdp op  as well as nested_matmul backward?

File: test/test_nestedtensor.py [hunk 3]
```diff
@@ -993,7 +993,7 @@ def test_matmul(self, device, dtype):
             r"matmul: Expected size for the 1st dimension of 2nd input tensor to be: [0-9]+ but got: [0-9]+.",
             lambda: torch.matmul(nt1, nt0)
         )
-        # error case: incompatible generalized batch size
+        # error case: incompatible (wrong) batch sizes that shouldn't even broadcast?
```
Comment (cpuhrsch) [inline test/test_nestedtensor.py:996]: Yes, this shouldn't work since matmul(torch.randn((2, 2, 4), torch.randn((3, 4, 6)) doesn't work.

**Review by cpuhrsch — APPROVED** [review]
Great! Also like that you cleaned up the error messages.

**Review by drisspg — APPROVED** [review]
LG!

File: aten/src/ATen/native/nested/NestedTensorMath.cpp [hunk 4]
```diff
@@ -877,53 +880,41 @@ Tensor matmul_nested(const Tensor& self, const Tensor& mat2) {
...
+    "matmul: Nested tensors cannot be matrix multiplied, last dimension of self has sizes",
+    self_dim_size,
+    "second last dimension of mat2 has sizes",
+    mat2_dim_size);
+  // Construct output size from input sizes
+  Tensor output_sizes = self_sizes.clone();
+  // The last entry in every row of output_sizes should be last column of mat2_sizes
+  output_sizes.index_put_({at::indexing::Slice(), -1}, mat2_sizes.select(1, -1).clone());
+
+  auto self_padded = nested_to_padded_tensor(self_contig, 0.);
```
Comment (drisspg) [inline aten/src/ATen/native/nested/NestedTensorMath.cpp:913]: I think you need need the dispatcher call  `at::nested_to_padded_tensor`

**Pull request closed without merging**